### ISSUE TITLE
Plugin not firing

### ISSUE THREAD

**Issue by hugopeek** [SEDAdigital/XRouting#28]
Hi guys,

I've been using XRouting on many installs now without issues, but for the latest one the plugin fails to execute.

The &xrouting-debug=1 setting doesn't work when appended to the URL and activated in system settings, so it's not a matter of the plugin not *rerouting* properly. It simply doesn't fire. I reinstalled the plugin and checked if the proper system events were enabled, but that seems to be in order (OnContextRemove, OnContextSave, OnHandleRequest, OnSiteRefresh).

I then disabled the plugin and installed ContextRouter instead. This immediately routed to the correct subdomain, so all context and .htaccess settings seem to be correct. The error log is silent (on this one.. ;)).

That's not really much to go on, so just PM me if you'd like to debug it in the manager yourself. I'm good for now with ContextRouter.

**Comment by christianseel** [comment]
Hey Hugo,
which MODX version is it? Did you try adding a `$modx->log()` to the beginning of the plugin to see if it's actually executed. Also inside the `case 'OnHandleRequest':` condition?

I could imagine that an other plugin is doing something wired/bad OnHandleRequest and ContextRouter might use a different event?

**Comment by hugopeek** [comment]
Hi Chris,

Sorry for the delayed response. It's MODX 2.5.4. I added the log lines inside the plugin, and the plugin does actually reach the OnHandleRequest case.

I tried disabling all other plugins that execute OnHandleRequest, but nothing.. The subdomain is routing just fine on localhost. That's an exact duplicate of the site, except for the base_url, site_url and http_host context settings of course.

The PHP version on both servers is also the same (5.6). The only difference is that the live server uses Apache instead of Nginx and has SSL enabled.

**Comment by christianseel** [comment]
Hey, did you find out in which line the plugin escaped and didn't continue?

**Comment by hugopeek** [comment]
It just happened again. This time on a local install, so we can happily debug the $#!+ out of it!

I couldn't pinpoint at which line the plugin fails.. Messages are reaching the log from every position I placed them.

But I figured out how to display the debug info this time (needed to place a / between the url and &xrouting-debug=1), and it doesn't show the affected context in the array of hosts. There are 4 other contexts there and they all route just fine. And also this time, installing ContextRouter and disabling Xrouting results in successful rerouting to the 'invisible' 5th context.

How to proceed?

**Comment by hugopeek** [comment]
AAAAAAAARGH!! I found it. Turns out, the contexts that were not routing properly had their `base_url` context setting set as `base_href` instead. Must have glanced over these a million times!!!

And weird that ContentRouter doesn't care as much about that setting apparently.. It was only until I saw the base_href setting pop up in Git after a Gitify extract that I realized my mistake.

Anyway, thanks for the help and sorry to have bothered you!
